scripts/cluster_network.py

```diff
@@ -160,6 +162,22 @@ def distribute_clusters(n, n_clusters, solver_name=None):
     assert n_clusters >= len(N) and n_clusters <= N.sum(), \
         "Number of clusters must be {} <= n_clusters <= {} for this selection of countries.".format(len(N), N.sum())
 
+    if focus_weights is not None:
+
+        total_focus = sum(list(focus_weights.values()))
+
+        assert total_focus <= 1.0, "The sum of focus weights must be less than 1."
+        
+        for country, weight in focus_weights.items():
+            L[country] = weight
```

Comment:
```suggestion
            L[country] = weight / len(L[country])
```
Like this you don't have to worry about how many subnetworks lie in the country and L always sums up to zero. So the assertion check below is not necessary

Reply:
No, the idea is for the focus weight is to specify the share of clusters to be allocated to that country (e.g. DE: 0.5 should result in 50% of the nodes allocated to Germany and the remainder is distributed proportional to mean load for all the other countries which are not listed). Your suggestion would distort this.

Reply:
Okay, quick question: 
What if Denmark has 2 synchronous zones and we want to give Denmark 40% of all nodes. Then `L["DK"]` would include two entries. When we give them both a value of 0.4, the total of Denmark would have a weight of 80%. By dividing by `len(L["DK"])`, we guarantee that the resulting total weight of denmark is 40%. 

Am I missing a point here?

Reply:
Ah, I was confused because of the assertion check which should still be done, but you are right.

Reply:
changed in https://github.com/PyPSA/pypsa-eur/commit/7df2b76f9daeb8d2775b833edf486c770e46c682